zerver/actions/streams.py

```diff
@@ -787,9 +788,11 @@ def bulk_add_subscriptions(
     altered_user_dict: dict[int, set[int]] = defaultdict(set)
     altered_guests: set[int] = set()
     altered_streams_dict: dict[UserProfile, set[int]] = defaultdict(set)
+    stream_to_subscribers: dict[Stream, set[UserProfile]] = defaultdict(set)
     for sub_info in subs_to_add + subs_to_activate:
         altered_user_dict[sub_info.stream.id].add(sub_info.user.id)
         altered_streams_dict[sub_info.user].add(sub_info.stream.id)
+        stream_to_subscribers[sub_info.stream].add(sub_info.user)
```

Comment:
I think it might be better to only add the user ID to the list if it `is_active` here -- saves putting large structures into a new set.